### DISCUSSION TITLE
Crosshair: Compatibility with Timeline

### DISCUSSION THREAD

**Discussion by mattp-** [f5/unovis#115] [Q&A]
the crosshair component as it stands today only operates on the x axis; this doesn't really make sense for a timeline; or at least it usually wont if you have multiple parallel items. having it be rotatable for the y axis would be useful.

**Comment by rokotyan** [comment]
Hi @mattp- 

We have the horizontal version of Crosshair in our pipeline, it's not our highest priority though. 

Can you please explain how would you like to use it with Timeline? We usually integrate them like this:
![Screen Recording 2023-01-18 at 1 17 38 PM](https://user-images.githubusercontent.com/755708/213297004-4a21ffb5-7ba7-490e-89e5-de86373bb5c2.gif)

**Comment by mattp-** [comment]
hi there, yep that looks like what i was looking for. mainly I was actually just looking for a workaround for https://github.com/f5/unovis/issues/103 to get my tooltip to appear. I thought I had it with crosshair until I realized it was not the right axis :)

**Comment by rokotyan** [comment]
@mattp- In that case, you'll need to set `snapToData` to `false` in Crosshair's config, and provide your own `getCircles` function for getting the position for the Crosshair circles. It's a bit tricky, but here is an example:

```ts
function findDataEntriesAt (x: number): SampleTimelineDatum[] {
  return data.filter(d => (d.time <= x) && ((d.time + d.duration) >= x))
}

function getCircles (x: number, data: SampleTimelineDatum[]): CrosshairCircle[] {
  return findDataEntriesAt(x).map((d, i) => ({
    y: 11 + i * 22 - scrollTop, // If your Timeline is scrollable you'll also need to store its current scrollTop position
    color: d.color,
  }))
}

// Optional: saving the scrollTop position and re-rendering the Crosshair
// Set this function via the `onScroll` config property of Timeline
function onScroll (top: number): void {
  setScrollTop(top)
  crosshairRef.component.render(0)
}
```
